sdk/keyvault/keyvault-certificates/src/core/challengeBasedAuthenticationPolicy.ts

```diff
@@ -49,34 +76,42 @@ export class ChallengeBasedAuthenticationPolicy extends BaseRequestPolicy {
     nextPolicy: RequestPolicy,
     options: RequestPolicyOptions,
     private credential: TokenCredential,
-    private tokenCache: AccessTokenCache
+    private tokenCache: AccessTokenCache,
+    private challengeCache: AuthenticationChallengeCache
   ) {
     super(nextPolicy, options);
   }
 
-  private parseWWWAuthenticate(www_authenticate: string): string {
+  private parseWWWAuthenticate(www_authenticate: string): {
+    authorization: string,
+    resource: string
+  } {
+    const returnValue = {
+      authorization: "",
+      resource: ""
+    };
     // Parses an authentication message like:
     // ```
     // Bearer authorization="some_authorization", resource="https://some.url"
     // ```
-    let authenticateArray = www_authenticate.split(" ");
-
-    // Remove the "Bearer" piece
-    delete authenticateArray[0];
+    let spaceSep = www_authenticate.split(" ");
```

Comment:
Since we remove quotes later, are we sure that splitting on space here is safe? Are there any cases where what is returned is a space inside of quotes?

Reply:
I believe there is no known instance in which we will receive spaces in between the keys and the values, or in the values. As far as I'm aware, it's always like `stuff <key>=<value>, <key>=<value>`, like how you described it in this function's comments: `Bearer authorization="some_authorization", resource="https://some.url"`. cc: @schaabs